core/src/main/java/io/questdb/InotifyFileEventNotifier.java

```diff
@@ -0,0 +1,42 @@
+package io.questdb;
+
+import io.questdb.std.str.Path;
+
+public final class InotifyFileEventNotifier implements FileEventNotifier {
+
+    private final long filewatcherPtr;
+    private boolean closed;
+
+    public InotifyFileEventNotifier(CharSequence dirPath) {
+        try (Path p = new Path()) {
+            p.of(dirPath).$();
+            this.filewatcherPtr = setup(p.ptr());
```

Comment:
the `setup` returns 0 in case of parameter error, we should at least have an assert here